code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/InputBoxNotificationBuilder.kt

```diff
@@ -0,0 +1,297 @@
+        val cacheService = ServiceProvider.getInstance().cacheService
+            ?: throw NotificationConstructionFailedException("Cache service is null, input box push notification will not be constructed.")
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Building an input box template push notification."
+        )
+        val packageName = context.packageName
+        val smallLayout = RemoteViews(packageName, R.layout.push_template_collapsed)
+        val expandedLayout = RemoteViews(packageName, R.layout.push_template_expanded)
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val channelIdToUse: String = notificationManager.createNotificationChannelIfRequired(
+            context,
+            pushTemplate.channelId,
+            pushTemplate.sound,
+            pushTemplate.getNotificationImportance(),
+            pushTemplate.isFromIntent
+        )
+
+        // create the notification builder with the common settings applied
+        val notificationBuilder = AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            channelIdToUse,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            R.id.basic_expanded_layout
+        )
+
+        // get push payload data. if we are handling an intent then we know that we should be building a feedback received notification.
+        val imageUri =
+            if (!pushTemplate.isFromIntent) pushTemplate.imageUrl else pushTemplate.feedbackImage
+        val pushImage = PushTemplateImageUtil.downloadImage(cacheService, imageUri)
+
+        if (pushImage != null) {
+            expandedLayout.setImageViewBitmap(R.id.expanded_template_image, pushImage)
+        } else {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "No image found for input box push template."
+            )
+            expandedLayout.setViewVisibility(R.id.expanded_template_image, View.GONE)
+        }
+
+        val expandedBodyText =
+            if (pushTemplate.isFromIntent) pushTemplate.feedbackText else pushTemplate.expandedBodyText
+        val collapsedBodyText =
+            if (pushTemplate.isFromIntent) pushTemplate.feedbackText else pushTemplate.body
+        smallLayout.setTextViewText(R.id.notification_title, pushTemplate.title)
+        smallLayout.setTextViewText(R.id.notification_body, collapsedBodyText)
+        expandedLayout.setTextViewText(R.id.notification_title, pushTemplate.title)
+        expandedLayout.setTextViewText(
+            R.id.notification_body_expanded, expandedBodyText
+        )
+
+        // add an input box to capture user feedback if the push template is not from an intent
+        if (pushTemplate.isFromIntent || pushTemplate.inputBoxReceiverName.isNullOrEmpty()) {
```

Comment:
I thought  the `inputBoxReceiverName` should be required by default for all the input box style notifications based on the spec. When can this be null?

Reply:
you are correct, it is required. the `isNullOrEmpty` check can be removed.